diff --git a/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi b/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
index 5d0d5a316d19..4cc914d885fa 100644
--- a/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
+++ b/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
@@ -126,6 +126,12 @@
 		/* On Low speed expansion */
 			label = "LS-SPI0";
 			status = "okay";
+			/* cs-gpios = <&msmgpio 18 0>; */
+			spidev@0 {
+				compatible = "spidev";
+				spi-max-frequency = <1000000>;	/* 1MHz */
+				reg = <0>;
+			};
 		};
 
 		leds {
diff --git a/arch/arm64/boot/dts/qcom/msm8916-pins.dtsi b/arch/arm64/boot/dts/qcom/msm8916-pins.dtsi
index 610668a37c4e..66b9a4b692fe 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-pins.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916-pins.dtsi
@@ -209,6 +209,7 @@
 		};
 		pinmux_cs {
 			function = "gpio";
+			/* function = "blsp5_spi5"; */
 			pins = "gpio18";
 		};
 		pinconf {
@@ -227,6 +228,7 @@
 	spi5_sleep: spi5_sleep {
 		pinmux {
 			function = "gpio";
+			/* function = "blsp5_spi5"; */
 			pins = "gpio16", "gpio17", "gpio18", "gpio19";
 		};
 		pinconf {
