From a4d722c1739764b1935ec2013d09971e55509c09 Mon Sep 17 00:00:00 2001
From: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
Date: Wed, 7 Jun 2017 07:21:30 +0200
Subject: [PATCH 3/5] ASoC: codecs: msm8916-wcd-digital: set default mclk rate

Signed-off-by: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
---
 sound/soc/codecs/msm8916-wcd-digital.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/codecs/msm8916-wcd-digital.c b/sound/soc/codecs/msm8916-wcd-digital.c
index f690442af8c9..e061ee2752fc 100644
--- a/sound/soc/codecs/msm8916-wcd-digital.c
+++ b/sound/soc/codecs/msm8916-wcd-digital.c
@@ -189,6 +189,8 @@
 #define TXN_DMIC_CTL_CLK_SEL_DIV6		0x3
 #define TXN_DMIC_CTL_CLK_SEL_DIV16		0x4
 
+#define DEFAULT_MCLK_RATE 9600000
+
 #define MSM8916_WCD_DIGITAL_RATES (SNDRV_PCM_RATE_8000 | \
 				   SNDRV_PCM_RATE_16000 | \
 				   SNDRV_PCM_RATE_32000 | \
@@ -876,6 +878,9 @@ static int msm8916_wcd_digital_probe(struct platform_device *pdev)
 		return ret;
 	}
 
+	/* set mclk default mclk rate */
+	clk_set_rate(priv->mclk, DEFAULT_MCLK_RATE);
+
 	ret = clk_prepare_enable(priv->mclk);
 	if (ret < 0) {
 		dev_err(dev, "failed to enable mclk %d\n", ret);
-- 
2.11.0

