diff --git a/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi b/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
index 5d0d5a316d19..6a131e28075a 100644
--- a/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
+++ b/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
@@ -400,7 +400,7 @@
 				data-lanes = <0 2>;
 				qcom,settle-cnt = <0xe>;
 				remote-endpoint = <&ov5645_ep>;
-				status = "disabled";
+				status = "okay";
 			};
 		};
 		port@1 {
@@ -436,7 +436,7 @@
 		vdda-supply = <&camera_vdda_2v8>;
 		vddd-supply = <&camera_vddd_1v5>;
 
-		status = "disabled";
+		status = "okay";
 
 		port {
 			ov5645_ep: endpoint {
